src/components/Icon/FontIcon/FontIcon.tsx

```diff
@@ -0,0 +1,53 @@
+import React, { AriaRole, forwardRef } from "react";
+import classNames from "classnames";
+import VibeComponentProps from "../../../types/VibeComponentProps";
+import VibeComponent from "../../../types/VibeComponent";
+
+interface FontIconProps extends VibeComponentProps {
+  onClick?: (event: React.MouseEvent<HTMLElement>) => void;
+  "aria-label"?: string;
+  tabIndex?: number;
+  icon?: string | React.Component | null;
+  role?: AriaRole;
+  "aria-hidden"?: boolean;
+}
+
+const FontIcon: VibeComponent<FontIconProps, HTMLElement> = forwardRef(
+  (
+    {
+      id,
+      className,
+      onClick,
+      "aria-label": iconLabel,
+      tabIndex,
+      icon: Icon,
+      role = "img",
+      "aria-hidden": ariaHidden,
+      "data-testid": dataTestId
+    },
+    iconRef
+  ) => {
+    const isIconFunction = typeof Icon === "function";
```

Comment:
Are React components functions? If not, we should adjust types.

I think that React has a utility function for checking if something is a React component or not.

Reply:
It should be only a function component but i am afraid of breaking changes in weird cases (for example creating a custom renderer in a use callback hook... not sure it will count as function component)... developers really abuse this icon prop and I prefer to be safe